docs/products/opensearch/howto/oidc-authentication.rst

```diff
@@ -0,0 +1,91 @@
+
+   * **Refresh limit count**: Maximum number of unrecognized JWT key IDs allowed within 10 seconds. Enter the value for the Refresh Limit Count parameter. The default value is 10.
+   * **Refresh limit window (ms)**: This is the interval, measured in milliseconds, during which the system will verify unrecognized JWT key IDs. Enter the value for the Refresh Limit Window parameter. The default value is 10,000 (10 seconds).
+
+6. Select **Enable**  to complete the setup and activate the configuration.
+
+SSO authentication and authorization sequence
+-------------------------------------------------
+
+Learn how Aiven for OpenSearch uses OpenID Connect for secure Single Sign-On (SSO) with robust authentication and role-based authorization.
+
+1. **User Access Request**: A user tries to access the OpenSearch Dashboard provided by their Aiven for OpenSearch service.
+
+2. **Aiven Authentication Check**: Aiven's service checks whether the user already has an active session. If not, it prepares to initiate the SSO process.
+
+3. **Redirect to IdP**: Aiven for OpenSearch redirects the user to the configured OpenID Connect Identity Provider for authentication.
+
+4. **User Authentication**: At the IdP interface, the user provides their credentials. This could be a username and password, a one-time pin, or another form of authentication supported by the IdP.
+
+5. **Generation of ID Token**: Once the user is successfully authenticated, the IdP generates and sends an ID token (JWT) to Aiven. This token contains claims about the user's identity and possibly additional information.
+
+6. **Token Verification**: Aiven for OpenSearch verifies the ID token to ensure it's legitimately issued by the trusted IdP. It validates the token's signature, checks its expiration time, and may assess other token attributes.
+
+7. **Extraction of User Details**: After validating the token, Aiven extracts the user's information, using claims from the JWT, to understand the user's identity and roles.
+
+8. **Authorization Process**: With the user's identity verified, Aiven for OpenSearch references the roles or permissions, either embedded within the ID token or fetched separately from the IdP. Using these roles, Aiven determines what resources or actions the user can access within the OpenSearch Dashboard.
+
+9. **Access Decision**: Depending on the user's roles and permissions, Aiven for OpenSearch either grants or denies access to the OpenSearch Dashboard or specific features within it.
+
+10. **Session Creation**: If the user is authorized, Aiven establishes a session for the user in the OpenSearch Dashboard. This means the user won't need to re-authenticate for a specific duration, even if they navigate away and return shortly after.
+
+11. **Session End**: After a defined period of inactivity or upon user logout, Aiven terminates the session. For subsequent access, the user would be prompted to re-authenticate through the SSO process.
+
+
```

Comment:
I would add section that would direct to Opensearch security documentation for assigning them to proper roles (unfortunately the official documentation is pretty bad regarding this). Not sure how much explaining we should do though.